### ISSUE TITLE
checking out conflict in subdir fails (with the current main branch)

### ISSUE THREAD

**Issue by yuja** [martinvonz/jj#2165]
## Steps to Reproduce the Problem

```sh
for i in 0 1; do
    jj new root -m $i
    mkdir -p sub
    echo $i > sub/file
done
RUST_BACKTRACE=1 jj new 'description(0)' 'description(1)'
```

## Actual Behavior

```
thread 'main' panicked at 'unexpected conflict entry in diff at "sub/file"', lib/src/working_copy.rs:1253:29
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: jj_lib::working_copy::TreeState::update::{{closure}}
             at /home/yuya/work/2023/jj/lib/src/working_copy.rs:1253:29
   3: jj_lib::working_copy::TreeState::update
             at /home/yuya/work/2023/jj/lib/src/working_copy.rs:1286:13
   4: jj_lib::working_copy::TreeState::check_out
             at /home/yuya/work/2023/jj/lib/src/working_copy.rs:1179:21
   5: jj_lib::working_copy::LockedWorkingCopy::check_out
             at /home/yuya/work/2023/jj/lib/src/working_copy.rs:1578:21
   6: jj_lib::working_copy::WorkingCopy::check_out
             at /home/yuya/work/2023/jj/lib/src/working_copy.rs:1524:21
   7: jj_cli::cli_util::update_working_copy
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:1922:21
   8: jj_cli::cli_util::WorkspaceCommandHelper::update_working_copy
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:1346:21
   9: jj_cli::cli_util::WorkspaceCommandHelper::finish_transaction
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:1400:13
  10: jj_cli::cli_util::WorkspaceCommandTransaction::finish
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:1534:9
  11: jj_cli::commands::cmd_new
             at /home/yuya/work/2023/jj/cli/src/commands/mod.rs:2351:5
  12: jj_cli::commands::run_command
             at /home/yuya/work/2023/jj/cli/src/commands/mod.rs:3743:36
  13: core::ops::function::FnOnce::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
  14: core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
  15: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/alloc/src/boxed.rs:1993:9
  16: jj_cli::cli_util::CliRunner::run_internal
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:2699:9
  17: jj_cli::cli_util::CliRunner::run
             at /home/yuya/work/2023/jj/cli/src/cli_util.rs:2716:22
  18: jj::main
             at /home/yuya/work/2023/jj/cli/src/main.rs:18:5
  19: core::ops::function::FnOnce::call_once
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

## Specifications

- Platform: Linux
- Version: 0.8.0-79291a3ca4ab6009e2ce0c6b4bf36d6c5b973565

**Comment by martinvonz** [comment]
That's gotta be something I've broken. I'll look into it later. Thanks for letting me know.

**Comment by yuja** [comment]
Fixed by 598cfcb8
